pytorch_lightning/overrides/data_parallel.py

```diff
@@ -276,16 +289,22 @@ def _worker(i, module, input, kwargs, device=None):
 
                 # ---------------
                 # CHANGE
-                if module.training:
+                if module.running_stage == RunningStage.TRAINING:
```

Comment:
These changes might need to be reflected in the sharded override as well i think

Reply:
Great catch !